File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        nums.sort()

        longest_streak = 1
        curr_streak = 1

        for i in range(1,len(nums)):
            if nums[i] != nums[i-1]:
                if nums[i] == nums[i-1]+1:
                    curr_streak +=1
                else:
                    longest_streak = max(longest_streak, curr_streak)
                    curr_streak =1
        return max(longest_streak, curr_streak)
        

        

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int max = 0;

        for(int i = 0; i<nums.length; i++){
            set.add(nums[i]);
        }

        for(int num:set){
            if (!set.contains(num-1)){
                int curr = num;
                int streak = 1;

                while(set.contains(curr+1)){
                    streak+=1;
                    curr+=1;
                }

                max = Math.max(streak, max);

            }

        }

        return max;
        
    }
}